bpf/straceback-tailcall-bpf.c

```diff
@@ -192,7 +192,7 @@ int tracepoint__sys_enter(struct sys_enter_args *ctx)
 			// https://github.com/torvalds/linux/commit/9fd29c08e52023252f0480ab8f6906a1ecc9a8d5
 			switch (arg_len) {
 			case 0:
-				sc_cont.failed = true;
+				// Nothing to do
```

Comment:
I think it's not strictly required for the read case (the conditional above will never enter for the `read` case). Do you think we could brake something by changing that?


https://github.com/kinvolk/traceloop/blob/a4887a9514f3cd7f38b714670a4e031cc2d1031d/pkg/straceback/syscalls.go#L281-L283

Reply:
Right, for the `read()` case, it's not necessary since it will do the `bpf_probe_read()` in the `tracepoint__sys_exit()` function.

But semantically, I think that's the correct change, and it happens in the `write()` case, where the `bpf_probe_read()` is done in the `tracepoint__sys_enter()` function, when the tracee calls
```
write(fd, buf, 0);
```
At a first glance, it seems useless for a program to try to `write` zero bytes. But after reading `man 2 write`, I see there could be legitimate use cases for a program to do that in order to know why a previous write was partial (to distinguish between EDQUOT or EFBIG). In that case, I don't think traceloop should display an "Deref failed" error.